### ISSUE TITLE
Type on `object.__dir__` is incorrect.

### ISSUE THREAD

**Issue by ktbarrett** [python/typeshed#11313]
According to the documentation [`__dir__` must return a `Sequence`](https://docs.python.org/3/reference/datamodel.html#object.__dir__).

**Comment by hauntsaninja** [comment]
I think the Python docs can sometimes be a little imprecise about what "sequence" means. E.g., the actual code that `dir()` uses just calls `PySequence_List` on the result of `__dir__()`. `PySequence_List` is another good example of imprecise use of "sequence"; it's perfectly happy to take in an iterable. So everything always works with Iterable at runtime.

That said, I'm still fine with the proposed change if mypy_primer says everything is okay.

**Comment by srittau** [comment]
Indeed. Another alternative is to fix the Python docs to match the implementation. Maybe this is something we should try first? In any case, I think it's important to be precise here, for cases, where someone uses `__dir__` directly.

**Comment by srittau** [comment]
I suggest to move this issue to python/cpython and suggest to change the docs. Returning a set from `__dir__` makes a lot of sense to me.
